src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchResultTablePrinter.cs

```diff
@@ -14,18 +16,70 @@ internal class PackageSearchResultTablePrinter : IPackageSearchResultRenderer
         private ILoggerWithColor _loggerWithColor;
         private const int LineSeparatorLength = 40;
         private static readonly string SourceSeparator = new('*', LineSeparatorLength);
+        private PackageSearchVerbosity _verbosity;
+        private bool _exactMatch;
 
-        public PackageSearchResultTablePrinter(string searchTerm, ILoggerWithColor loggerWithColor)
+        private readonly string[] _minimalVerbosityTableHeader = { "Package ID", "Latest Version" };
+        private readonly string[] _minimalVerbosityTableHeaderForExactMatch = { "Package ID", "Version" };
+        private readonly int[] _minimalColumnsToHighlight = { 0 };
+
+        private readonly string[] _normalVerbosityTableHeader = { "Package ID", "Latest Version", "Owners", "Total Downloads" };
+        private readonly string[] _normalVerbosityTableHeaderForExactMatch = { "Package ID", "Version", "Owners", "Total Downloads" };
+        private readonly int[] _normalColumnsToHighlight = { 0, 2 };
+
+        private readonly string[] _detailedVerbosityTableHeader = { "Package ID", "Latest Version", "Owners", "Total Downloads", "Vulnerable", "Deprecation", "Project URL", "Description" };
+        private readonly string[] _detailedVerbosityTableHeaderForExactMatch = { "Package ID", "Version", "Owners", "Total Downloads", "Vulnerable", "Deprecation", "Project URL", "Description" };
+        private readonly int[] _detailedColumnsToHighlight = { 0, 2, 6, 7 };
```

Comment:
Due to different data being displayed to the customer for various verbosity levels, The `PackageSearchResultTablePrinter` class in its current form is handling too many responsibilities - it decides the table format based on verbosity, creates the table, populates it, and prints it.

1. Define Strategy interface
```cs
interface ITableFormatStrategy
{
    Table CreateTable(IEnumerable<IPackageSearchMetadata> results, bool exactMatch);
}
```
2. Implement strategies for each verbosity Level

```cs
class MinimalTableFormatStrategy : ITableFormatStrategy
{
    public Table CreateTable(IEnumerable<IPackageSearchMetadata> results, bool exactMatch)
    {
        // Implement logic for creating a minimal verbosity table
    }
}

class NormalTableFormatStrategy : ITableFormatStrategy
{
    public Table CreateTable(IEnumerable<IPackageSearchMetadata> results, bool exactMatch)
    {
        // Implement logic for creating a normal verbosity table
    }
}

class DetailedTableFormatStrategy : ITableFormatStrategy
{
    public Table CreateTable(IEnumerable<IPackageSearchMetadata> results, bool exactMatch)
    {
        // Implement logic for creating a detailed verbosity table
    }
}
```

4. Create factory to get appropriate strategy

```cs
class TableFormatStrategyFactory
{
    public static ITableFormatStrategy GetStrategy(PackageSearchVerbosity verbosity)
    {
        switch (verbosity)
        {
            case PackageSearchVerbosity.Minimal:
                return new MinimalTableFormatStrategy();
            case PackageSearchVerbosity.Normal:
                return new NormalTableFormatStrategy();
            case PackageSearchVerbosity.Detailed:
                return new DetailedTableFormatStrategy();
            default:
                throw new ArgumentException("Invalid verbosity level");
        }
    }
}
```

6. Refactor PackageSearchResultTablePrinter

```cs
class PackageSearchResultTablePrinter : IPackageSearchResultRenderer
{
    // Existing fields...

    public PackageSearchResultTablePrinter(string searchTerm, ILoggerWithColor loggerWithColor, PackageSearchVerbosity verbosity, bool exactMatch)
    {
        // Constructor implementation...
    }

    public void Add(PackageSource source, IEnumerable<IPackageSearchMetadata> completedSearch)
    {
        _loggerWithColor.LogMinimal(SourceSeparator);
        _loggerWithColor.LogMinimal($"Source: {source.Name} ({source.SourceUri})");

        ITableFormatStrategy strategy = TableFormatStrategyFactory.GetStrategy(_verbosity);
        Table table = strategy.CreateTable(completedSearch, _exactMatch);
        table.PrintResult(_searchTerm, _loggerWithColor);
    }

    public void Add(PackageSource source, string error)
    {
        // Error handling implementation...
    }

    // Implement other methods...
}
```

Reply:
@Nigusu-Allehu  - Feel free to address this comment in a follow-up PR. cc @kartheekp-ms

Reply:
I agree with Arturo to address this comment in a follow-up PR. 

Nigusu, please feel free to create a tracking issue, add a link here and resolve the conversation.

Reply:
Fixed here : https://github.com/NuGet/NuGet.Client/pull/5602